ee/tuf/autoupdate.go

```diff
@@ -749,3 +771,40 @@ func (ta *TufAutoupdater) cleanUpOldErrors() {
 		)
 	}
 }
+
+// collectAndSetEnrollmentDetails collects the runtime enrollment details for the
```

Comment:
Is this duplicated with the one in run?

Reply:
I think it'd be reasonable to add a function to `pkg/osquery/enrollment_details.go` that looks like `CollectAndSetEnrollmentDetails(ctx context.Context, k types.Knapsack)` so we don't have to duplicate it here + below `runLauncher` 🙂 . (Or maybe `CollectAndSetEnrollmentDetails(ctx context.Context, k types.Knapsack, collectTimeout time.Duration, collectRetryInterval time.Duration)`?)